packages/@o3r/components/builders/metadata-check/helpers/config-metadata-comparison.helper.ts

```diff
@@ -0,0 +1,65 @@
+import { ComponentConfigOutput } from '@o3r/components';
+import { MetadataComparator } from '@o3r/extractors';
+
+/**
+ * Interface describing a config migration element
+ */
+export interface MigrationConfigData {
+  libraryName: string;
+  configName: string;
+  propertyName?: string;
+}
+
+/**
+ * Comparator used to compare one version of config metadata with another
+ */
+export class ConfigMetadataComparator implements MetadataComparator<ComponentConfigOutput> {
+  /**
+   * Returns an array of config metadata from a metadata file.
+   * To be easily parseable, the properties will be split in separate items of the array.
+   * e.g. : [{ library: '@o3r/demo', properties: [{name : 'property1', type: 'string'}, {name : 'property2', type: 'number'}] }]
+   * will become :
+   * [{ library: '@o3r/demo', properties: [{name : 'property1', type: 'string'}] }, { library: '@o3r/demo', properties: [{name : 'property2', type: 'number'}] }]
+   *
+   * @param content Content of a migration metadata files
+   */
+  public getArray(content: ComponentConfigOutput[]): ComponentConfigOutput[] {
+    return content.reduce((acc, config) => {
+      if (config.properties.length) {
+        const propertiesConfigs = config.properties.map((property) => {
+          return {
+            ...config,
+            properties: [property]
+          };
+        });
+        acc.push(...propertiesConfigs);
+      } else {
+        acc.push(config);
+      }
+      return acc;
+    }, [] as ComponentConfigOutput[]);
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public getName(config: ComponentConfigOutput): string {
+    return `${config.library}#${config.name}` + config.properties.length ? `-${config.properties[0].name}` : '';
```

Comment:
why the `[0]`?

Reply:
Because we have the split the configuration to have only one property each time